plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AllFilesTable.java

```diff
@@ -0,0 +1,172 @@
+                pagesBuilder.appendIntegerBigintMap(dataFile.nanValueCounts());
+            }
+            if (checkNonNull(dataFile.lowerBounds(), pagesBuilder)) {
+                pagesBuilder.appendIntegerVarcharMap(dataFile.lowerBounds().entrySet().stream()
+                        .collect(toImmutableMap(
+                                Map.Entry<Integer, ByteBuffer>::getKey,
+                                entry -> Transforms.identity(idToTypeMapping.get(entry.getKey())).toHumanString(
+                                        Conversions.fromByteBuffer(idToTypeMapping.get(entry.getKey()), entry.getValue())))));
+            }
+            if (checkNonNull(dataFile.upperBounds(), pagesBuilder)) {
+                pagesBuilder.appendIntegerVarcharMap(dataFile.upperBounds().entrySet().stream()
+                        .collect(toImmutableMap(
+                                Map.Entry<Integer, ByteBuffer>::getKey,
+                                entry -> Transforms.identity(idToTypeMapping.get(entry.getKey())).toHumanString(
+                                        Conversions.fromByteBuffer(idToTypeMapping.get(entry.getKey()), entry.getValue())))));
+            }
+            if (checkNonNull(dataFile.keyMetadata(), pagesBuilder)) {
+                pagesBuilder.appendVarbinary(Slices.wrappedBuffer(dataFile.keyMetadata()));
+            }
+            if (checkNonNull(dataFile.splitOffsets(), pagesBuilder)) {
+                pagesBuilder.appendBigintArray(dataFile.splitOffsets());
+            }
+            if (checkNonNull(dataFile.equalityFieldIds(), pagesBuilder)) {
+                pagesBuilder.appendIntegerArray(dataFile.equalityFieldIds());
+            }
+            pagesBuilder.endRow();
+        });
+
+        return pagesBuilder.build();
+    }
+
+    protected static boolean checkNonNull(Object object, PageListBuilder pagesBuilder) {
+        if (object == null) {
+            pagesBuilder.appendNull();
+            return false;
+        }
+        return true;
+    }
+
+    protected static Map<Integer, Type> getIcebergIdToTypeMapping(Schema schema) {
+        ImmutableMap.Builder<Integer, Type> icebergIdToTypeMapping = ImmutableMap.builder();
+        for (Types.NestedField field : schema.columns()) {
+            AllFilesTable.populateIcebergIdToTypeMapping(field, icebergIdToTypeMapping);
+        }
+        return icebergIdToTypeMapping.buildOrThrow();
+    }
+
+    private static void populateIcebergIdToTypeMapping(Types.NestedField field, ImmutableMap.Builder<Integer, Type> icebergIdToTypeMapping) {
+        Type type = field.type();
+        icebergIdToTypeMapping.put(field.fieldId(), type);
+        if (type instanceof Type.NestedType) {
+            type.asNestedType().fields().forEach(child -> AllFilesTable.populateIcebergIdToTypeMapping(child, icebergIdToTypeMapping));
+        }
+    }
+
+    @Override
+    public Distribution getDistribution() {
```

Comment:
I'd move these up above the private methods so all the `@Override`s are together like they were in `FilesTable`